graph/connected_components_with_dsu.cpp

```diff
@@ -0,0 +1,53 @@
+#include <iostream>
+#include <vector>
+#include <set>
+
+int N;  // denotes number of nodes;
+int parent[1000001];
+int siz[1000001];
```

Comment:
same for this array too

Reply:
Thanks for pointing out. Fixed.
All arrays are now dynamically allocated.